### ISSUE TITLE
ALSA fail on set_rate with 5.19 kernel

### ISSUE THREAD

**Issue by mervinbeng** [HEnquist/camilladsp#225]
Hello,

I see this error since updating to 5.19 kernel. This capture is on a linux-lqx kernel. It is identical on a standard arch linux kernel.

On downgrading to a 5.18.x kernel there no error on starting camilladsp.

Thanks,
Mervin

----

# camilladsp ~/drc/camilladsp/pw_peq2.cfg
2022-08-18 05:55:01.162862 INFO [src/bin.rs:711] CamillaDSP version 1.0.1
2022-08-18 05:55:01.162899 INFO [src/bin.rs:712] Running on linux, x86_64
2022-08-18 05:55:01.164769 ERROR [src/bin.rs:344] **Playback error: ALSA function 'snd_pcm_hw_params_set_rate' failed with error 'EINVAL: Invalid argument'**
2022-08-18 05:55:01.179267 INFO [src/processing.rs:37] Playback thread has already stopped.

$ uname -a
Linux ### 5.19.1-lqx1-1-lqx #1 ZEN SMP PREEMPT_DYNAMIC Sat, 13 Aug 2022 22:26:34 +0000 x86_64 GNU/Linux

$ head drc/camilladsp/pw_peq2.cfg
---
devices:
  samplerate: 48000
  chunksize: 512
  silence_timeout: 0
  enable_rate_adjust: false
  enable_resampling: false
  resampler_type: BalancedAsync
  capture:

**Comment by HEnquist** [comment]
Can you try starting again with debug logging level? It should give much more info on supported parameter values while setting up the device.

**Comment by mervinbeng** [comment]
Thanks Henrik. Using debug I've narrowed down the issue:

Pre-5.19 kernel - no issues starting camilladsp on any clock rate providing it matches the pipewire-pulse clock rate. It's fixed because I use the Pro (pipwire-jack) device.

5.19 kernel - on startup even though pw-top shows 48000, camilladsp debug shows the clock is 44100. Then is cannot set the clock to 48000, which is what I have on the config for camilladsp. I've since tested 96000 (both pipewire and camilladsp), and it works.

I will report this to the pipewire team. The issue seems somewhere between piewire and the kernel. In the example below, pipewire.conf default clock rate = 48000, but seemed to be set at 44100. When set to 96000, it gives 96000, so that it my workaround for the moment.

...
2022-08-18 09:04:34.039240 DEBUG [src/alsadevice.rs:412] Playback: supported channels, min: 2, max: 12, list: [2, 12]
2022-08-18 09:04:34.039262 DEBUG [src/alsadevice.rs:413] Playback: setting channels to 2
2022-08-18 09:04:34.039281 DEBUG [src/alsadevice.rs:417] Playback: **supported samplerates: Discrete([44100])**
2022-08-18 09:04:34.039295 DEBUG [src/alsadevice.rs:418] **Playback: setting rate to 48000**
2022-08-18 09:04:34.039329 ERROR [src/bin.rs:344] Playback error: ALSA function 'snd_pcm_hw_params_set_rate' failed with error 'EINVAL: Invalid argument'
2022-08-18 09:04:34.039344 DEBUG [src/bin.rs:349] Error while starting, release barrier
...
